gno.land/pkg/sdk/vm/msgs.go

```diff
@@ -12,6 +12,15 @@ import (
 	"github.com/gnolang/gno/tm2/pkg/std"
 )
 
+type Format string
+
+const (
+	FormatMachine Format = "machine" // Default machine representation
+	FormatString         = "string"  // Single string representation
+	FormatJSON           = "json"    // XXX: EXPERIMENTAL, only supports primitive types for now
```

Comment:
```suggestion
	FormatJSON           = "json"    // XXX: EXPERIMENTAL, only supports primitive types for now
	FormatJSONP          = "jsonp"    // XXX: EXPERIMENTAL, only supports primitive types for now
```

I'm for adding right now the `jsonp` alternative. Btw `jsonp` takes an extra argument (the callback name).

Reply:
Why jsonp? (who would use jsonp in 2025?)

Reply:
I will add `jsonp` later; it complicates the query by supporting extra args. Let's focus on having a working JSON output . We could add this kind of parameters by supporting URL queries on the query path later.